app/handlers/GuaranteeBalanceResponseHandler.scala

```diff
@@ -62,8 +63,13 @@ class GuaranteeBalanceResponseHandler @Inject() (
         logger.warn(s"[GuaranteeBalanceResponseHandler][processResponse]Failed to process Response. BalanceRequestFunctionalError: $errors")
         technicalDifficulties()
       case Left(failureResponse) =>
-        logger.warn(s"[GuaranteeBalanceResponseHandler][processResponse]Failed to process Response: $failureResponse")
-        technicalDifficulties()
+        if (failureResponse.status.equals(TOO_MANY_REQUESTS)) {
+          Future.successful(Redirect(controllers.routes.RateLimitController.onPageLoad()))
```

Comment:
I'm not a totally fluent Scala dev, but I'm wondering if this would be better handled as part of the match/case: 
case Left(failureResponse) if failureResponse.status.equals(TOO_MANY_REQUESTS) => 409 case
case Left(failureResponse) => tech difficulties case.